src/core/public/application/application_service.tsx

```diff
@@ -67,8 +67,9 @@ const getAppUrl = (mounters: Map<string, Mounter>, appId: string, path: string =
   return appendAppPath(appBasePath, path);
 };
 
-const getAppDeepLinkPath = (mounters: Map<string, Mounter>, appId: string, deepLinkId: string) => {
-  return mounters.get(appId)?.deepLinkPaths[deepLinkId];
+const getAppDeepLinkPath = (app: App<any>, appId: string, deepLinkId: string) => {
+  const flattenedLinks = flattenDeepLinks(app.deepLinks);
+  return flattenedLinks[deepLinkId];
```

Comment:
We're now computing the deeplinks paths on every call, but this is only called when navigating to an app, and the operation is not really expensive. This was better ihmo that keeping the app and the mounter in sync.